src/Facets/UnitFacet.sol

```diff
@@ -0,0 +1,220 @@
+    function startBridgeTokensViaUnit(
+        ILiFi.BridgeData memory _bridgeData,
+        UnitData calldata _unitData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, LibAsset.NULL_ADDRESS) // only allow native asset
+        onlyAllowDestinationChain(_bridgeData, LIFI_CHAIN_ID_HYPERCORE)
+    {
+        _verifySignature(_bridgeData, _unitData);
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _unitData);
+    }
+
+    /// @notice Performs a swap before bridging via Unit
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _unitData Data specific to Unit
+    function swapAndStartBridgeTokensViaUnit(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        UnitData calldata _unitData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        onlyAllowSourceToken(_bridgeData, LibAsset.NULL_ADDRESS) // only allow native asset
+        onlyAllowDestinationChain(_bridgeData, LIFI_CHAIN_ID_HYPERCORE)
+    {
+        _validateSwapOutputIsNative(_swapData);
+        _verifySignature(_bridgeData, _unitData);
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _unitData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Validates that the final swap output is native asset
+    /// @param _swapData Array of swap data
+    function _validateSwapOutputIsNative(
+        LibSwap.SwapData[] calldata _swapData
+    ) internal pure {
+        if (
+            !LibAsset.isNativeAsset(
+                _swapData[_swapData.length - 1].receivingAssetId
+            )
+        ) {
+            revert InvalidCallData();
+        }
+    }
+
+    /// @dev Contains the business logic for the bridge via Unit
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _unitData Data specific to Unit
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        UnitData calldata _unitData
+    ) internal {
+        if (block.chainid == 1) {
+            // ethereum mainnet
+            if (_bridgeData.minAmount < 0.05 ether) {
+                revert InvalidAmount();
+            }
+        } else if (block.chainid == 9745) {
```

Comment:
use constants instead of hardcoded chainIds, e.g. CHAIN_ID_ETHEREUM / CHAIN_ID_PLASMA